app/models/targeting.rb

```diff
@@ -0,0 +1,29 @@
+class Targeting < ActiveRecord::Base
+
+  belongs_to :user
+  belongs_to :bookmark
+
+  has_many :targeting_hosts, :dependent => :destroy
+  has_many :hosts, :through => :targeting_hosts
+  has_many :template_invocations, :dependent => :destroy
```

Comment:
should this be `has_many`? I think `has_one` is more appropriate, if you want to reuse same collection we should reuse bookmark/host collection I think and always copy out the query from it